### ISSUE TITLE
Support multi-level configuration

### ISSUE THREAD

**Issue by kzantow** [anchore/fangs#50]
There have been multiple requests for additions related to configuration in the CLI tools, and we should see if there are things that `fangs` can do to help facilitate the asks. Perhaps among other things, we might like to:

* support specifying multiple configuration files
* support reading multiple configuration files in the lookup paths
* support profiles to override or add to certain variables in a similar manner

One of the issues is: https://github.com/anchore/grype/issues/2009, where a user would like to have one configuration that they share among many repositories and a second configuration that a repository itself may contain.

In order to best serve users, we need to determine a path forward. Some solutions included:

* use some kind of `include` or `import` directive in a configuration file to pull in information from another file
* read and merge the configurations, where things like lists get appended to
* read and merge configurations, where fields are overridden

It is quickly apparent that a user may have different desires and certain fields may make sense to override vs. append in different cases, and having a mix of this behavior might be confusing.

It is possible that on a per-field value, developers could _tag a field with the behavior_ they would like to see of merge or override. An example might look like:
```
type ScanOptions struct {
    excludeDirectories []string `yaml:"exclude" mapstructure:"exclude" merge:"append"`
}
```

 It is possible a user might want to specify this, too. Maybe something like:

```yaml
# global.config
name: joe's laptop
scan-options:
    +exclude: /tmp
```

```yaml
# repo.config
name: some-project
scan-options:
    +exclude: ./target
```

More research is needed to determine how other tools might be doing this and if we could follow any established patterns, or if Viper has any particular support for this type of thing ([Viper documentation](https://github.com/spf13/viper#reading-config-files) says "Viper can search multiple paths, but currently a single Viper instance only supports a single configuration file.").

**Comment by kzantow** [comment]
A go library that seems to support some more complicated yaml-based configuration that may at least help to provide some prior art: https://carvel.dev/ytt/

**Comment by kzantow** [comment]
I thought about this for a bit, and I think I have a reasonable behavior that should be both very flexible and minimize unexpected behavior, I implemented profiles because they are somewhat key to making this all sane. I've implemented this behavior in: https://github.com/anchore/fangs/pull/51. The main thing that could be confusing is that in many cases users would like to merge lists together, but not always, and while it may be dependent on the context it could be more confusing why it happens for some fields and not others.

Here is an explanation of an approach I think minimize confusion and maximizes flexibility:
* If a user specifies one or more configuration files explicitly, those files are used and they must exist, no implicit files are used. the files are used in priority order from first specified
* If no configuration files are specified, _all_ the configurations found in implicit locations are used, in same priority order that was used to find a file: current directory `config` or `.appname/config`, home directory `.config`/`.appname/config`, XDG dirs. this is a change in behavior that may be considered breaking so there's a config option to stop at the first file found, like it does today
* environment variables override all entries in config files and profiles
* when multiple configuration files are used: slices are _appended_
* when profiles are used, all values are overridden, including slices

This may sound overly complicated, but I think it results in the most expected and desired behaviors, while allowing just about anything a user wants fairly easily with profiles. If we think about it in terms of priority and appending vs overriding: **multiple files append, profiles override, priority is always first wins, profiles have higher priority**. Some examples:

Example 1: a grype user specifies some ignore rules in their home directory, projects specify some additional ignore rules.

`$HOME/.grype.yaml` -- contains user's ignore rules
`.grype.yaml` -- in current directory has more ignore rules

If the user did not specify a config file, it would implicitly find: `.grype.yaml`, `$HOME/.grype.yaml` in that order, the set of ignore rules would contain what was specified in BOTH files, with `$HOME/.grype.yaml` being at the end of the list.

If the user runs `grype -c .grype.yaml`, it would _only_ use that config file and not implicitly look up any. If the user runs: `grype -c .grype.yaml -c $HOME/.grype.yaml`, it would have the same behavior as the case where these files were implicitly found, above, where `.grype.yaml` takes precedence, and the ignore rules have the `$HOME/.grype.yaml` appended at the end of the list.

Example 2: the same files exist, a `profile` named `no-ignores` exists in either file, with an empty list of ignore rules. The user runs `grype --config-profile no-ignores`, the empty ignore list _replaces_ the non-profile ignores, easily allowing non-appending override behavior.

Example 3: the same files exist, but a user wants to append MORE ignores without modifying either file. The user can create a third file, let's say `.grype-ignores.yaml`, and run `grype -c .grype.yaml -c $HOME/.grype.yaml -c .grype-ignores.yaml` to append all the lists together.

...

I think if the profiles had appending behavior, this would be rather unexpected and would prevent use cases like the `no-ignores` above. If neither had appending behavior many of the desired use cases wouldn't work, or it was specified on a per-field basis it would get even more confusing very quickly what is expected. I think by sticking to these rules, it is reasonably easy to understand what's going on.
